app/categorization/llm_categorization.py

```diff
@@ -1,169 +1,71 @@
-    try:
-        float(s)
-        return True
-    except ValueError:
-        return False
-
-
-def is_iso8601(s: str) -> bool:
-    try:
-        datetime.fromisoformat(s)
-        return True
-    except ValueError:
-        return False
-
-
-def is_european_decimal(s: str) -> bool:
-    s = s.strip()  # Remove leading and trailing whitespace
-
-    if s.count(",") == 1:
-        if s.index(",") > 0 and s.index(",") < len(s) - 1:
-            return True
-
-    return False
-
-
-# def is_bounded(s:str)->bool: yet to be added
-# def is_years yet to be added
-
-
-def AgeFormat(result_dict: Dict[str, str], r: str, key: str) -> Dict[str, Any]:
-    value = result_dict[key].strip()  # Ensure no leading/trailing whitespace
-    numbers_list_str = value.split()
-    Age_l = [num_str.strip() for num_str in numbers_list_str]
-    Fvar: str = "Unknown"
-
-    for num_str in Age_l:
-        if is_integer(num_str):
-            Fvar = "integervalue"
-            break
-        elif is_float(num_str):
-            Fvar = "floatvalue"
-            break
-        elif is_iso8601(num_str):
-            Fvar = "iso8601"
-            break
-        elif is_european_decimal(num_str):
-            Fvar = "europeandecimalvalue"
-            break
-        # 2 more conditions yet to be added
-
-    output: Dict[str, str] = {
-        "TermURL": "nb:Age",
-        "Format": Fvar,
-    }
-    return output
-
-
-def llm_invocation(result_dict: Dict[str, str]) -> Dict[str, Any]:
 
-    # Initialize model
+def AssessmentTool(key: str, value: str) -> Optional[Dict[str, str]]:
     llm = ChatOllama(model="gemma")
+    questionAssessmentTool = f"Is the {key}:{value} an assessment tool"
+    chainAssessmentTool = AssessmentToolPrompt | llm
+    llm_response_Assessment = chainAssessmentTool.invoke(
+        {"column": key, "content": value, "question": questionA}
+    )
+    reply = str(llm_response_Assessment)
+    if "yes" in reply.lower():
+        output = {"TermURL": "nb:Assessment"}
+        print(json.dumps(output))
+        return output
+    else:
+        print(key)
+        print("not in data model")
+        return None
 
-    # Create prompt template
-    prompt = PromptTemplate(
-        template="""Given the column data {column}: {content},
-    determine the category and give only the category name as output
-
-Examples:
-1. Input: "participant_id: sub-01 sub-02 sub-03"
-Output: Participant_IDs
-
-2. Input: 'pheno_age: ["34,1", "35,3", "NA", "39,0", "22,1",
-"23,2", "21,1", "22,3", "42,5", "43,2"]'
-Output: Age
-
-3. Input: "session_id: ses-01 ses-02"
-Output: Session_IDs
-
-4. Input: "pheno_sex : ["F", "F", "M", "M", "missing",
-"missing", "F", "F", "M", "M"]"
-Output: Sex
 
-5. Input: "pheno_sex : ["1", "2", "1", "2", "missing", "missing"]"
-Output: Sex
+def llm_diagnosis_assessment(key: str, value: str) -> Optional[Dict[str, str]]:
+    resultDiagnosis = Diagnosis(key, value)
+    if resultDiagnosis:
+        return resultDiagnosis
```

Comment:
Does this logic catch columns that a neither Diagnosis nor Assessment Tool (such as socioeconomic state or level of education)? As far as I can see it does "If it is not a diagnosis it is an assessment tool.", right?

Reply:
first its checked for diagnosis if not then for  assessment inside the assessment function I have kept an option for "not in data model" currently there is just a print statement I am planning to store it in a dictionary but I am not sure about global declaration of dict in production codes so I am researching on other ways.